src/generic.jl

```diff
@@ -44,94 +44,85 @@ __eltype(::Base.EltypeUnknown, a) = _eltype(typeof(first(a)))
 
 # Generic column-wise evaluation
 
+@deprecate colwise!(r::AbstractArray, metric::PreMetric, a, b) zipwise!(r, metric, a, b)
+
 """
-    colwise!(r::AbstractArray, metric::PreMetric, a, b)
+    zipwise!(r::AbstractArray, metric::PreMetric, a, b)
```

Comment:
TBH I don't find `zipwise` to be a super natural name. This exists only because `map!` only accepts arrays? We could define `map!` methods that work on any iterators even if the fallback `map!` method doesn't.

Reply:
Yea, this was just a proposal, since colwise wouldn't make much sense anymore. I realized the whole `map!` issue only after I did the "zipwise" commit. I guess `map!` doesn't exist for general iterators because they may not have a known length, but one could indeed work around that.

Reply:
I can't see a good reason for `map!` not to support arbitrary iterators as inputs given that `map` supports them. The docstring mentions "collection" without any details. We could file an issue in Julia to discuss this.

Reply:
https://github.com/JuliaLang/julia/issues/38344